<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.82.1" />


<title>Use pipe with R - Luyao Kang</title>
<meta property="og:title" content="Use pipe with R - Luyao Kang">



  








<link href='//cdn.bootcss.com/highlight.js/9.11.0/styles/github.min.css' rel='stylesheet' type='text/css' />



<link rel="stylesheet" href="../../../../../css/fonts.css" media="all">
<link rel="stylesheet" href="../../../../../css/main.css" media="all">

<link rel="stylesheet" href="../../../../../css/custom.css">

<script async src="https://use.fontawesome.com/32c3d13def.js"></script>
<link rel="stylesheet" href="../../../../../css/academicons.css" media="all">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="../../../../../" class="nav-logo">
    <img src="../../../../../images/logo.png"
         width="60"
         height="50"
         alt="Luyao Kang">
  </a>

  <ul class="nav-links">
    
    
    
    <li class=""><a href="../../../../../"> Home </a></li>
    
    <li class=""><a href="../../../../../about/"> About </a></li>
    
    <li class=""><a href="../../../../../en/"> Blog </a></li>
    
    <li class=""><a href="../../../../../cn/"> 中文 </a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    
    <span class="article-duration">3 min read</span>
    
    <h1 class="article-title">Use pipe with R

</h1>


<div class="article-date">
  <span> Luyao Kang ·   2021/12/21</span>
  <span class="article-toolbar">
    
    <a href="../../../../../en/index.xml" type="application/rss+xml" target="_blank"><i class="fa fa-rss" aria-hidden="true" title="RSS feed"></i></a>
    
    <a href="https://twitter.com/home?status=Use&#43;pipe&#43;with&#43;R&#43;%2Fen%2F2021%2F12%2F21%2Fuse-pipe-with-r%2F&#43;via&#43;%40loyal_ECO" target="_blank"><i class="fa fa-twitter" aria-hidden="true" title="Share via Twitter"></i></a>
    <a href="http://service.weibo.com/share/share.php?content=utf-8&amp;title=Use&#43;pipe&#43;with&#43;R&#43;%40Kangluyao&amp;url=%2Fen%2F2021%2F12%2F21%2Fuse-pipe-with-r%2F" target="_blank"><i class="fa fa-weibo" aria-hidden="true" title="分享到新浪微博"></i></a>
    <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank"><i class="fa fa-cc" aria-hidden="true" title="Attribution-NonCommercial-ShareAlike 4.0 International"></i></a>
    
    
    
    <a href="https://github.com/kangluyao/kangluyao.github.io/content/en%5c2021-12-21-pipe-in-r.md" target="_blank"><i class="fa fa-pencil-square-o" aria-hidden="true" title="Suggest an edit of this page"></i></a>
    
    </span>
</div>



    
    <div class="article-content">
      <h1 id="set-work-directory">Set work directory</h1>
<pre><code class="language-{r}" data-lang="{r}">setwd('E:/thermokast_lakes/water_microbes/')
</code></pre><h1 id="loading-packages">Loading packages</h1>
<pre><code class="language-{r}" data-lang="{r}">ipak &lt;- function(pkg){
  new.pkg &lt;- pkg[!(pkg %in% installed.packages()[, &quot;Package&quot;])]
  if (length(new.pkg)) 
    install.packages(new.pkg, dependencies = TRUE)
  sapply(pkg, require, character.only = TRUE, warn.conflicts = FALSE)
}
pkgs &lt;- c('ape', 'Biostrings', 'phyloseq',               
              'tidyr', 'dplyr', 'ggplot2')
ipak(pkgs)
</code></pre><h1 id="conduct-a-phyloseq-project">Conduct a phyloseq project</h1>
<pre><code class="language-{r}" data-lang="{r}">
#read in metadata
metadata &lt;- read.csv(&quot;./meta_analysis/data/meta_data/sample_data.csv&quot;,
                     header = T, row.names = 1)

#read in otu table
meta.otu.table &lt;- read.csv(&quot;./meta_analysis/data/meta_data/meta_otu_table.csv&quot;,
                           header = T, row.names = 1, stringsAsFactors = F)
meta.otu.table &lt;- as.matrix(meta.otu.table)

#read in taxonomy
meta.taxonomy &lt;- read.csv(&quot;./meta_analysis/data/meta_data/meta_taxonomy.csv&quot;,sep=&quot;,&quot;,row.names=1)
meta.taxonomy &lt;- as.matrix(meta.taxonomy)

# read in tree
meta.phy.tree &lt;- read_tree(&quot;./meta_analysis/data/meta_data/meta_tree.nwk&quot;)

#read in represent dna sequences
meta.ref.seqs &lt;- readDNAStringSet(file = &quot;./meta_analysis/data/meta_data/meta_ref_seqs.fasta&quot;,
                                  format = &quot;fasta&quot;, nrec = -1L, skip = 0L, seek.first.rec = FALSE, use.names = TRUE)
meta.otu.table &lt;- otu_table(meta.otu.table, taxa_are_rows = TRUE)
meta.tax.table &lt;- tax_table(meta.taxonomy)
meta.table &lt;- sample_data(metadata)
meta_physeq &lt;- phyloseq(meta.tax.table, meta.otu.table, meta.table, meta.phy.tree, meta.ref.seqs)
meta_physeq
</code></pre><pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 10250 taxa and 320 samples ]
sample_data() Sample Data:       [ 320 samples by 38 sample variables ]
tax_table()   Taxonomy Table:    [ 10250 taxa by 8 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 10250 tips and 10227 internal nodes ]
refseq()      DNAStringSet:      [ 10250 reference sequences ]
</code></pre>
<h1 id="determine-the-alpha-diversity">Determine the alpha diversity</h1>
<pre><code class="language-{r}" data-lang="{r}">diversity &lt;- estimate_richness(meta_physeq, measures = c(&quot;Chao1&quot;, 'Shannon', 'Simpson'))
meta_diversity &lt;- cbind(diversity, Region = metadata$Region,
                        Site = metadata$Site,  Sitegroup1 = metadata$Sitegroup1)
</code></pre><pre><code class="language-{r}" data-lang="{r}">knitr::kable(meta_diversity[1:5, 1:7])
</code></pre><table>
<thead>
<tr>
<th style="text-align:left"></th>
<th style="text-align:right">Chao1</th>
<th style="text-align:right">se.chao1</th>
<th style="text-align:right">Shannon</th>
<th style="text-align:right">Simpson</th>
<th style="text-align:left">Region</th>
<th style="text-align:left">Site</th>
<th style="text-align:left">Sitegroup1</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">ERR4650773</td>
<td style="text-align:right">44.0000</td>
<td style="text-align:right">9.625787</td>
<td style="text-align:right">1.661566</td>
<td style="text-align:right">0.742798</td>
<td style="text-align:left">Pan-Arctic</td>
<td style="text-align:left">A</td>
<td style="text-align:left">North_East Siberia</td>
</tr>
<tr>
<td style="text-align:left">ERR4650774</td>
<td style="text-align:right">45.1000</td>
<td style="text-align:right">6.481545</td>
<td style="text-align:right">1.660527</td>
<td style="text-align:right">0.726742</td>
<td style="text-align:left">Pan-Arctic</td>
<td style="text-align:left">A</td>
<td style="text-align:left">North_East Siberia</td>
</tr>
<tr>
<td style="text-align:left">ERR4650775</td>
<td style="text-align:right">127.2308</td>
<td style="text-align:right">23.000918</td>
<td style="text-align:right">2.428847</td>
<td style="text-align:right">0.802124</td>
<td style="text-align:left">Pan-Arctic</td>
<td style="text-align:left">A</td>
<td style="text-align:left">North_East Siberia</td>
</tr>
<tr>
<td style="text-align:left">ERR4650776</td>
<td style="text-align:right">121.1765</td>
<td style="text-align:right">17.164433</td>
<td style="text-align:right">2.619853</td>
<td style="text-align:right">0.841466</td>
<td style="text-align:left">Pan-Arctic</td>
<td style="text-align:left">A</td>
<td style="text-align:left">North_East Siberia</td>
</tr>
<tr>
<td style="text-align:left">ERR4650777</td>
<td style="text-align:right">105.5652</td>
<td style="text-align:right">10.045071</td>
<td style="text-align:right">2.966866</td>
<td style="text-align:right">0.900248</td>
<td style="text-align:left">Pan-Arctic</td>
<td style="text-align:left">A</td>
<td style="text-align:left">North_East Siberia</td>
</tr>
</tbody>
</table>
<h1 id="manipulating-and-analyzing-data-with-dplyr-and-tidyr">Manipulating and analyzing data with dplyr and tidyr</h1>
<pre><code>tp_alpha_div &lt;-  meta_diversity %&gt;% 
  filter(Region == 'Tibetan Plateau') %&gt;%
  dplyr::select(-c(2)) %&gt;%
  group_by(Region, Site, Sitegroup1) %&gt;%
  dplyr::summarise(across(, mean, na.rm = TRUE)) %&gt;%
  tidyr::gather(diversity, value, -c('Region', 'Site', 'Sitegroup1')) %&gt;%
  dplyr::mutate(Sitegroup1 = factor(Sitegroup1, levels = c('MD', 'QL', 'QZR')))
knitr::kable(tp_alpha_div[1:6, ])
</code></pre><pre><code>`summarise()` regrouping output by 'Region', 'Site' (override with `.groups` argument)
</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left">Region</th>
<th style="text-align:left">Site</th>
<th style="text-align:left">Sitegroup1</th>
<th style="text-align:left">diversity</th>
<th style="text-align:right">value</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z1</td>
<td style="text-align:left">QL</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">496.7214</td>
</tr>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z10</td>
<td style="text-align:left">QZR</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">415.7837</td>
</tr>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z11</td>
<td style="text-align:left">QZR</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">376.7769</td>
</tr>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z12</td>
<td style="text-align:left">QZR</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">387.1121</td>
</tr>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z13</td>
<td style="text-align:left">QZR</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">227.1032</td>
</tr>
<tr>
<td style="text-align:left">Tibetan Plateau</td>
<td style="text-align:left">Z14</td>
<td style="text-align:left">QZR</td>
<td style="text-align:left">Chao1</td>
<td style="text-align:right">517.1068</td>
</tr>
</tbody>
</table>
<h1 id="plot-with-ggplot2">Plot with ggplot2</h1>
<pre><code>alpha_region_plot &lt;- ggplot(tp_alpha_div, aes(x = Sitegroup1, y = value, fill = Sitegroup1)) +
  geom_violin(trim=T, width=0.5, aes(fill = Sitegroup1), colour = &quot;#000000&quot;) +
  scale_fill_manual(values= c('#d95f02', '#1b9e77', 'red')) +
  geom_boxplot(width=0.1, fill=&quot;white&quot;, colour = &quot;#000000&quot;) +
  labs(x = 'Sitegroup1', y = 'Alpha diversity') +
  facet_wrap(~diversity, scales = 'free') +
  theme_bw() +
  theme(strip.text = element_text(size = 14),
        axis.line = element_line(colour = &quot;black&quot;),
        axis.title = element_text(size = 16, colour = 'black'),
        axis.text = element_text(size = 14, colour = 'black'),
        axis.text.x = element_text(colour='black', size = 14, 
                                   angle = 45, hjust = 1),
        panel.grid = element_blank(), legend.position = 'none')
options(repr.plot.width = 6, repr.plot.height = 4, repr.plot.res = 300)
print(alpha_region_plot)
</code></pre><p><img src="../../../../../images/2021_12_21.png" alt=""></p>

    </div>
    

<nav id="article-nav">
    
    <a href="../../../../../en/2021/06/29/submit-sequence-data-to-ncbi/" id="article-nav-older" class="article-nav-link-wrap">
      <div class="article-nav-title"><i class="fa fa-arrow-circle-left" aria-hidden="true"></i> How to submit sequence data to NCBI</div>
    </a>
    

    
    <a href="../../../../../en/2022/10/05/markdown-emoji-markup/" id="article-nav-newer" class="article-nav-link-wrap">
      <div class="article-nav-title">Markdown emoji markup <i class="fa fa-arrow-circle-right" aria-hidden="true"></i></div>
    </a>
    
</nav>
<script type="text/javascript">
document.addEventListener('keyup', function(e) {
  if (e.target.nodeName.toUpperCase() != 'BODY') return;
  var url = false;
  if (e.which == 37) {  
    
    url = '\/en\/2021\/06\/29\/submit-sequence-data-to-ncbi\/';
    
  } else if (e.which == 39) {  
    
    url = '\/en\/2022\/10\/05\/markdown-emoji-markup\/';
    
  }
  if (url) window.location = url;
});
</script>



  </article>

  
<section id="comments">
  <div id="disqus_thread"></div>
  <script>
  
  var disqus_config = function () {
    this.page.url = "https:\/\/kangluyao.github.io\/" + location.pathname;
  };
  
  (function() {
   var inIFrame = function() {
      var iframe = true;
      try { iframe = window.self !== window.top; } catch (e) {}
      return iframe;
    };
    if (inIFrame()) return;
    var d = document, s = d.createElement('script');
    s.src = '//luyao-en.disqus.com/embed.js'; s.async = true;
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>





</main>

      <footer class="footer">
        <ul class="footer-links">
          
          <li><a href="mailto:lykang365@gmail.com"><i class="fa fa-envelope" aria-hidden="true" title="Mail to Luyao"></i><span class="sr-only">Email</span></a></li>
          <li><a href="../../../../../"><i class="ai ai-google-scholar" aria-hidden="true" title="Google Scholar"></i><span class="sr-only">Google Scholar</span></a></li>
          
          <li><a href="https://github.com/kangluyao"><i class="fa fa-github" aria-hidden="true" title="Github"></i><span class="sr-only">Github</span></a></li>
          <li><a href="https://twitter.com/loyal_ECO"><i class="fa fa-twitter" aria-hidden="true" title="Twitter"></i><span class="sr-only">Twitter</span></a></li>
          
          <li><a href="http://creativecommons.org/licenses/by-nc-sa/4.0/"><i class="fa fa-cc" aria-hidden="true" title="Attribution-NonCommercial-ShareAlike 4.0 International"></i><span class="sr-only">Attribution-NonCommercial-ShareAlike 4.0 International</span></a></li>
          <li><a href="../../../../../"><i class="fa fa-copyright" aria-hidden="true" title="Copyright"></i> 2021 - 2022</a></li>
        </ul>
      </footer>
    </div>
    
    <script async src="../../../../../js/center-img.js"></script>
    
    <script async src="../../../../../js/right-quote.js"></script>
    
    <script async src="../../../../../js/no-highlight.js"></script>
    
    <script async src="../../../../../js/fix-footnote.js"></script>
    
    

    



<script src="//cdn.bootcss.com/highlight.js/9.11.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



    
<script src="../../../../../js/math-code.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [['$','$'], ['\\(','\\)']],
    processEscapes: true
  }
});
</script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>


    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-28785465-3', 'auto');
	
	ga('send', 'pageview');
}
</script>
  </body>
</html>

